src/RoleObjects/GridData.js

```diff
@@ -65,71 +82,345 @@ export default class GridData extends TableData {
     return this._reactProps['aria-readonly'];
   }
 
-  onKeyDown(event) {
```

Comment:
This wasn't follow WAI-ARIA practices and made assumptions about option attributes always being set in a particular way.  Hence the large refactor.  The refactor also breaks apart keydown handling into multiple functions so that treegrid can override parts as needed, and since it would have been way too long as a single function.